contextproject-ms3/src/main/java/contextproject/audio/SkipAudioProcessor.java

```diff
@@ -0,0 +1,59 @@
+package contextproject.audio;
+
+import be.tarsos.dsp.AudioEvent;
+import be.tarsos.dsp.AudioProcessor;
+
+import contextproject.models.Track;
+
+public class SkipAudioProcessor implements AudioProcessor {
+  
+  private double secondsToSkip;
+  private SkipAudioProcessorCallback callback;
+  private boolean hasFinished;
+  private boolean stopped = false;
+
+  public interface SkipAudioProcessorCallback {
+    public void onFinished();
+  }
+  
+  /**
+   * Because of the way MP3 streaming works, we can't use dispatcher.skip().
+   * Instead, this processor silently goes through the stream and cancels
+   * the rest of the dispatcher chain until the desired point is reached.
+   * 
+   */
+  public SkipAudioProcessor(double secondsToSkip, SkipAudioProcessorCallback callback) {
+    // TODO: get starting point based on track beatgrid
+    this.callback = callback;
+    this.secondsToSkip = secondsToSkip;
+  }
+  
+  @Override
+  public boolean process(AudioEvent audioEvent) {
+    if (stopped) {
+      return true;
+    }
+    
+    if (audioEvent.getTimeStamp() > secondsToSkip) {
```

Comment:
This processor will be changed. Could you do:
```git checkout feature-beatmatch contextproject-ms3/src/main/java/contextproject/audio/SkipAudioProcessor.java```
, update your code and check if it still works? You'll probably need to call skipProcessor.notify() instead of stop() in onFinished.